pandas/tests/indexes/test_multi.py

```diff
@@ -2269,6 +2271,20 @@ def test_unique(self):
         exp = pd.MultiIndex.from_arrays([['a'], ['a']])
         tm.assert_index_equal(res, exp)
 
+        # GH #17896 - with level= argument
```

Comment:
try with named level, try on an already unique level as well. ideally even more corner cases.

Reply:
Added [here](https://github.com/pandas-dev/pandas/pull/17897/files#diff-be59f06a56750dffb20853248fb3b4e1R2281)

Reply:
make this a separate test and parametrize